src/password/tests.py

```diff
@@ -0,0 +1,335 @@
+        Tests the unexpected case where a user uses an email reset, but the username doesn't match the one retrieved
+        """
+        setup_test_environment()
+        client = Client()
+        username = 'bad_username'
+        new_password = 'newpass'
+        body = {
+            'token': os.environ.get('LOGIN_SEC_TOKEN'),
+            'username': username,
+            'password': new_password
+        }
+        response = client.post(
+            '/password/change',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        error = json.loads(response.content.decode('utf-8'))['errors']
+        self.assertTrue('login_error' in error)
+        self.assertEqual(response.status_code, 403)
+
+    @patch('cassy.verify_auth_token')
+    def test_password_change_with_admin_empty_username(self, cassy_auth):
+        """
+        Tests the case where an admin resets someone's password but with an empty or missing username
+        Mocking the admin authentication.
+        """
+        setup_test_environment()
+        client = Client()
+        cassy_auth.return_value = 'admin'
+        new_password = 'newpass'
+        auth_token = 'token'
+        body = {
+            'auth_token': auth_token,
+            'password': new_password
+        }
+        response = client.post(
+            '/password/change',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        error = json.loads(response.content.decode('utf-8'))['errors']
+        self.assertTrue('reset_error_username' in error)
+        cassy_auth.assert_called_once_with(auth_token)
+        self.assertEqual(response.status_code, 400)
+
+    def test_password_change_incorrect_old_pass(self):
+        """
+         Tests the case where user is logged in, requests a new password but supplies incorrect old password
+        """
+        setup_test_environment()
+        client = Client()
+        new_password = 'newpass'
+        body = {
+            'auth_token': os.environ.get('LOGIN_SEC_TOKEN'),
+            'old': 'badpass',
+            'password': new_password
+        }
+        response = client.post(
+            '/password/change',
+            data=json.dumps(body),
+            content_type='application/json'
+        )
+        error = json.loads(response.content.decode('utf-8'))['errors']
+        self.assertTrue('login_error' in error)
+        self.assertEqual(response.status_code, 403)
+
+    def test_password_change_using_get(self):
+        """
+         Tests the case where user is logged in, requests a new password but supplies incorrect old password
+        """
+        setup_test_environment()
+        client = Client()
+        response = client.get(
+            '/password/change'
+            + '?auth_token=' + os.environ.get('LOGIN_SEC_TOKEN')
+            + '&old=badpass'
+            + '&password=newpass'
+        )
+        self.assertEqual(response.status_code, 400)
```

Comment:
When the response code is an error, should also check the error code to be sure it's what we expect. See lines 163/164 for example.